### PR TITLE
Turns on python-experimental CI tests, original

### PR DESCRIPTION
**Pull request by spacether**
Turns on python-experimental CI tests

- pom.xml samples updated to indicate which node number corresponds to which set of samples
- circleci config file adjusted to allow a single node to use a different executor like docker for example

Reasons that I made these changes:
- the pyenv on the current machine is out of date and does not allow installation of new python versions >= 3.7
- to do that one would need build pyenv from scratch
- our current solution feels brittle and is not meeting the specific needs of the languages in a node. Everything is piled into one place
- How about we use docker images instead, built only for what we need? This PR demonstrates that by breaking out circleci node3 into a docker image with additional installations into it.

I have found a docker image which contains all python versions and will allow us to test python 2.7 all the way up to 3.9 which python-experimental needs. We will need to install the below into it:
- [DONE] maven
- [DONE] java jdk
- docker (to run the server)
- typescript/node for the other node3 tests

How about I make a dockerhub account for openapi generator which holds this new image?
If we need to switch to other CI sites in the future, using this image should be easier than installing all needed software onto one machine like we currently do.
Does this sound like a good plan to you?
What email account can I use to make the dockerhub acccount for openapi generator (I don't want it to be my personal one)?

## Help!
- I am having trouble installing docker in this `Linux b2caa1cb8dfd 5.11.0-1022-aws #23~20.04.1-Ubuntu SMP Mon Nov 15 14:03:19 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux` image

TODO:
- install docker in the running image
- run the server in the background of it
- turn the python-legacy samples tests back on
- install whatever typescript/node/angular and javascript need to run

### PR checklist
 
- [ ] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [ ] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [ ] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [ ] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (5.3.0), `6.0.x`
- [ ] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

File: CI/circle_parallel.sh [hunk 1]
```diff
@@ -57,8 +57,10 @@ elif [ "$NODE_INDEX" = "3" ]; then
   #cd Python-3.8.9
   #./configure --enable-optimizations
   #sudo make altinstall
+  pyenv --version
```
Comment (spacether) [inline CI/circle_parallel.sh:60]: The pyenv version that we have is `pyenv 1.1.5-2-g19cdbc8` and we need at least [1.2.21 or later](https://github.com/pyenv/pyenv/blob/master/CHANGELOG.md#1221) to install python 3.9.0 or higher. I am not seeing pyenv in apt-get
What would you like me to do here?
- Install all the pre-reqs and compile pyenv?
This is pretty involved per here: https://medium.datadriveninvestor.com/how-to-install-and-manage-multiple-python-versions-on-linux-916990dabe4b
- Or should we switch to a docker image? How would I do that as there are already 3 or 4 nodes
@wing328

File: CI/circle_parallel.sh [hunk 2]
```diff
@@ -16,14 +16,21 @@ function cleanup {
...
-  echo "Running node $NODE_INDEX to test 'samples.circleci' defined in pom.xml ..."
+if [ "$NODE_INDEX" = "0" ]; then
+  echo "Running node $NODE_INDEX to test 'samples.circleci.node$NODE_INDEX' defined in pom.xml ..."
+  #sudo update-java-alternatives -s java-1.7.0-openjdk-amd64
+  java -version
+
+  mvn --no-snapshot-updates --quiet verify -Psamples.circleci.node0 -Dorg.slf4j.simpleLogger.defaultLogLevel=error
+  mvn --no-snapshot-updates --quiet javadoc:javadoc -Psamples.circleci.node1 -Dorg.slf4j.simpleLogger.defaultLogLevel=error
+elif [ "$NODE_INDEX" = "1" ]; then
+  echo "Running node $NODE_INDEX to test 'samples.circleci.node$NODE_INDEX' defined in pom.xml ..."
```
Comment (spacether) [inline CI/circle_parallel.sh:27]: This code was moved from below up to here. Only the echoing of the node sample was updated.

**Comment by wing328** [comment]
Looks like the following works for installing python 3.9:
```
  # install pyenv latest version and python 3.6.3
  rm -Rf /opt/circleci/.pyenv
  curl -L https://github.com/pyenv/pyenv-installer/raw/master/bin/pyenv-installer | bash
  source ~/.bash_profile
  pyenv --version
  pyenv install --list
  pyenv install 3.6.3

  # install openssl 1.1.1 and python 3.9
  wget --no-check-certificate https://www.openssl.org/source/openssl-1.1.1g.tar.gz
  tar zxvf openssl-1.1.1g.tar.gz
  cd openssl-1.1.1g
  ./config --prefix=/home/circleci/openssl --openssldir=/home/circleci/openssl no-ssl2
  make
  make test
  make install
  cd ..
  export PATH=$HOME/openssl/bin:$PATH
  export LD_LIBRARY_PATH=$HOME/openssl/lib
  export LC_ALL="en_US.UTF-8"
  export LDFLAGS="-L /home/circleci/openssl/lib -Wl,-rpath,/home/circleci/openssl/lib"
  which openssl
  openssl version
  CPPFLAGS=-I$HOME/openssl/include LDFLAGS=-L$HOME/openssl/lib SSH=$HOME/openssl pyenv install -v 3.9.0
```

**Comment by spacether** [comment]
@wing328 thank you for your help with that!

**Comment by spacether** [comment]
This PR will be abandoned in place of https://github.com/OpenAPITools/openapi-generator/pull/11390
@wing328 I wasn't able to get your suggestion working

**Pull request closed without merging**